src/client-scopes/add/components/components.ts

```diff
@@ -0,0 +1,27 @@
+import type { FC } from "react";
+
+import type { ConfigPropertyRepresentation } from "@keycloak/keycloak-admin-client/lib/defs/authenticatorConfigInfoRepresentation";
+import { StringComponent } from "./StringComponent";
+import { BooleanComponent } from "./BooleanComponent";
+import { ListComponent } from "./ListComponent";
+import { RoleComponent } from "./RoleComponent";
+import { ScriptComponent } from "./ScriptComponent";
+
+export type ComponentProps = Omit<ConfigPropertyRepresentation, "type">;
+export const ComponentTypes: ReadonlyArray<string> = [
+  "String",
+  "boolean",
+  "List",
+  "Role",
+  "Script",
+];
+
+export type Components = typeof ComponentTypes[number];
+
+export const COMPONENTS: { [index in Components]: FC<ComponentProps> } = {
```

Comment:
We're using `FunctionComponent` in other places, let's keep that consistent.
```suggestion
export const COMPONENTS: {
  [index in Components]: FunctionComponent<ComponentProps>;
} = {
```

Reply:
`FC` is just shorthand for `FunctionComponent`, from react:
```typescript
type FC<P = {}> = FunctionComponent<P>;
```
but I'll change it